web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,77 @@
+/*
+    Copyright 2016 Siddharth Bhalgami <siddharth.bhalgami@techreceptives.com>
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+*/
+odoo.define('web_widget_image_webcam.webcam_widget', function(require) {
+"use strict";
+
+    var core = require('web.core');
+
+    core.form_widget_registry.get("image").include({
+
+        render_value: function () {
+            this._super();
+
+            var self = this,
+                web_cam = this.$el.find('.oe_form_binary_file_web_cam');
+
+            $(web_cam).click(function(){
+
+                // ::webcamjs:: < https://github.com/jhuckaby/webcamjs >
+                // Webcam: Set Custom Parameters
+                Webcam.set({
+                    width: 320,
+                    height: 240,
+                    dest_width: 320,
+                    dest_height: 240,
+                    image_format: 'jpeg',
+                    jpeg_quality: 90,
+
+                    /*
+                        :: Important Note about Chrome 47+ :: < https://github.com/jhuckaby/webcamjs#important-note-for-chrome-47 >
+                        Setting "force_flash" to "true" will always run in Adobe Flash fallback mode on Chrome, but it is not desirable.
+                    */
+                    force_flash: false,
+                    /*
+                        If "force_flash" is true, then this extra parameter "swfURL" is necessary.
+                        Remove comment tag from "swfURL" parameter.
+                    */
+                    // swfURL: '/web_widget_image_webcam/static/src/js/webcam.swf',
+                    flip_horiz: true,
+                    fps: 45,
+                });
+
+                // Init Webcam
+                Webcam.attach('#live_webcam');
+
+                $('#take_snap_btn').click(function(){
+                    Webcam.snap( function(data) {
+                        // Display Snap besides Live WebCam Preview
+                        document.getElementById('webcam_result').innerHTML = '<img src="'+data+'"/>';
```

Comment:
You're welcome. No need to answer "done" to every comment, just push the commit and they will disappear automatically :wink: